src/CoreBundle/Bridge/Chameleon/Module/NavigationTreeSingleSelect/NavigationTreeSingleSelect.php

```diff
@@ -113,27 +275,158 @@ private function getPortalTreeRootNodeId(): string
         return $portal->fieldMainNodeTree;
     }
 
-    private function createTreeDataModel(\TdbCmsTree $node, string $fieldName, $path = ''): BackendTreeNodeDataModel
+    private function createTreeDataModel(\TdbCmsTree $node, int $level, string $path = ''): BackendTreeNodeDataModel
     {
         $treeNodeDataModel = $this->backendTreeNodeFactory->createTreeNodeDataModelFromTreeRecord($node);
-        $children = $node->GetChildren(true);
+        $treeNodeDataModel->setName($this->translateNodeName($treeNodeDataModel->getName(), $node));
+        $this->setTypeAndAttributes($treeNodeDataModel, $node);
+
+        if ($level === 0) {
+            $treeNodeDataModel->setOpened(true);
+        }
 
-        $path = $this->addBreadcrumbHtmlForNode($node, $fieldName, $path);
+        //"primary-node-selection (from page)" starts with the active portal:
+        //   $level: 0 = portal, 1 = Navigation-Nodes, >1 = folder or page
 
+        //"portal-selection" and "portal-home-page-selection" and "portal-404-page-selection start with root node:
+        //   $level: 0 = rootNode (Website), 1 = portal, 2 = Navigation-Nodes, >2 folder or page
+        if (true === $this->isPortalSelectMode) {
+            if ($level !== 1) {
+                $treeNodeDataModel->setDisabled(true);
+                $treeNodeDataModel->addListHtmlClass('no-checkbox');
+            }
+        } elseif (true === $this->isPortalHomeNodeSelectMode) {
+            if ($level < 2) {
+                $treeNodeDataModel->setDisabled(true);
+                $treeNodeDataModel->addListHtmlClass('no-checkbox');
+            }
+        } else {  // primary-node-selection from page
+            if ($level <= 1) {
+                $treeNodeDataModel->setDisabled(true);
+                $treeNodeDataModel->addListHtmlClass('no-checkbox');
+            }
+        }
+
+        ++$level;
+        $children = $node->GetChildren(true);
         while ($child = $children->Next()) {
-            $childTreeNodeObj = $this->createTreeDataModel($child, $fieldName, $path);
+            $childTreeNodeObj = $this->createTreeDataModel($child, $level, $path);
             $treeNodeDataModel->addChildren($childTreeNodeObj);
         }
 
         return $treeNodeDataModel;
     }
 
-    protected function addBreadcrumbHtmlForNode(\TdbCmsTree $node, string $fieldName, string $path = ''): string
+
+    private function translateNodeName(string $name, \TdbCmsTree $node): string
```

Comment:
Isn't $node already loaded in the correct language?

Reply:
Yes, but with a fallback, if the current language was not available. In fallback case  the name isn't the current language. Then a translation sign will be added.